tasks/deploy/coordinators/reth.ts

```diff
@@ -0,0 +1,136 @@
+import { task, types } from "hardhat/config";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+dayjs.extend(duration);
+import { parseEther, toFunctionSelector } from "viem";
+import { z } from "zod";
+import { zAddress } from "../utils";
+import { DeployCoordinatorsBaseParams } from "./shared";
+
+export let zRETHCoordinatorDeployConfig = z.object({
+  reth: zAddress.optional(),
+});
+
+export type RETHCoordinatorDeployConfigInput = z.input<
+  typeof zRETHCoordinatorDeployConfig
+>;
+
+export type RETHCoordinatorDeployConfig = z.infer<
+  typeof zRETHCoordinatorDeployConfig
+>;
+
+export type DeployCoordinatorsRethParams = {
+  admin?: string;
+  overwrite?: boolean;
+};
+
+task("deploy:coordinators:reth", "deploys the RETH deployment coordinator")
+  .addOptionalParam("admin", "admin address", undefined, types.string)
+  .addOptionalParam(
+    "overwrite",
+    "overwrite deployment artifacts if they exist",
+    false,
+    types.boolean,
+  )
+  .setAction(
+    async (
+      { admin, overwrite }: DeployCoordinatorsRethParams,
+      {
+        deployments,
+        run,
+        network,
+        viem,
+        getNamedAccounts,
+        config: hardhatConfig,
+      },
+    ) => {
+      let artifacts = await deployments.all();
+      if (!overwrite && artifacts["HyperdriveFactory"]) {
+        console.log(`HyperdriveFactory already deployed`);
+        return;
+      }
```

Comment:
Is this right? I'd expect us to check the `RETHHyperdriveDeployerCoordinator` artifact to make sure we don't overwrite the coordinator. FWIW, I didn't see this in the `ezETH` script, so we may want to add that there.